### DISCUSSION TITLE
Unable to connect to TM1 Server via tm1py

### DISCUSSION THREAD

**Discussion by giesp** [cubewise-code/tm1py#989] [Q&A]
Hey,
I'm just (trying to) getting started with tm1py. I am already familiar with the REST API.
My problem is that I already fail at connecting to my TM1 instance. The error message I get is: "ValueError: No response returned from URL: 'https://192.168.1.144:8010/api/v1/Configuration/ProductVersion/$value'. Please double check your address and port number in the URL."
The check.py script for testing the connection returns the same error message.

The URL looks good to me and when I enter it in my REST client, the request succeeds (response 11.8.02200.2).  
I am testing against a simple demo machine (remote via VPN) with standard configuration and native authentication. I am trying to reach the demo database "SData". 

Any suggestions where I am going wrong?

**Comment by MariusWirtz** [comment]
Hi @giesp,

most likely something wrong with the URL. 

Can you run the script on the same machine with TM1? What happens then?

**Reply by giesp** [reply]
Hi @MariusWirtz,

Thank you for your reply!
In the meantime, I have done some research myself. I have not yet been able to find a final solution to the problem, but it seems to have something to do with the server configuration.

I have checked four test scenarios:
1. the request mentioned in the original post directly via tm1py - this does not work
2. the request generated by tm1py via my REST client - works.
3. send the request via tm1py to another server - works.
4. send the request in python directly via the requests library - does not work, but the error message is different:
SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1000)'))
I think an administrator should check the configuration or the certificates on the server.

Unfortunately, python is not installed on the TM1 server itself. So I cannot test on the server directly. After my tests, however, I am sure that it would work there.
